plugins/module_utils/cluster.py

```diff
@@ -84,6 +84,9 @@ def shutdown(
                 dict(forceShutdown=force_shutdown),
                 check_mode,
             )
+        # TimeoutError is handled in rest_client.create_record:
+        # except TimeoutError as e:
+        #       raise errors.ScaleComputingError(f"Request timed out: {e}")
         except ScaleComputingError as e:
             # To avoid timeout when there are a lot of VMs to shutdown
             if "Request timed out" in str(e):
```

Comment:
Here we also detect exception root cause from string, not from type. Are the more places with this pattern?

We should have a dedicated exception, like ScaleRequestTimedOut. Then you refactor the RestClient, and you need to check if there is any other module that needs to know about your new exception.

Reply:
But this error will never cause a problem right? The string is generated inside the client, so the string comparison will never fail due to formatting (like it would when we compared the string directly from the http response - in case of "connection refused")

Reply:
As discussed, I implemented ScaleTimeoutError in rest_client